File: master/backend/app/routes/recycle.py
"""
回收站路由
"""
from fastapi import APIRouter
from app.services.project_service import (
    list_recycle_projects,
    restore_project_from_recycle,
    purge_recycle_item
)

router = APIRouter(prefix="/api/recycle", tags=["recycle"])

@router.get("/projects")
def recycle_list():
    return list_recycle_projects()

@router.post("/projects/{pid}/restore")
def recycle_restore(pid: str):
    return restore_project_from_recycle(pid)

@router.delete("/projects/{pid}")
def recycle_delete(pid: str):
    return purge_recycle_item(pid)

File: master/frontend/src/api/recycle.ts
import client from './client';

export const listRecycleProjects = async (): Promise<{ items: Array<{ id: string; name: string; deleted_at: string }> }> => {
  const res = await client.get('/recycle/projects');
  return res.data;
};

export const restoreRecycleProject = async (pid: string): Promise<{ ok?: boolean }> => {
  const res = await client.post(`/recycle/projects/${pid}/restore`);
  return res.data;
};

export const deleteRecycleProject = async (pid: string): Promise<{ ok?: boolean }> => {
  const res = await client.delete(`/recycle/projects/${pid}`);
  return res.data;
};
